File: src/sensors/image/image.ts
import { Discoverable } from '../../discoverable';
import { ImageInfo } from './image-info';

/**
 * Implements an MQTT image for Home Assistant discovery
 */
export class Image extends Discoverable<ImageInfo> {
  /** Update the image URL */
  setUrl(imageUrl: string) {
    return this._state_helper(imageUrl);
  }
}

File: py-source/sensors/image.py
from ha_mqtt_discoverable import Discoverable
from ha_mqtt_discoverable.sensors import ImageInfo


class Image(Discoverable[ImageInfo]):
    """
    Implements an MQTT image for Home Assistant MQTT discovery:
    https://www.home-assistant.io/integrations/image.mqtt/
    """

    def set_url(self, image_url: str) -> None:
        """
        Update the camera state (image URL).

        Args:
            image_url (str): URL of the image to be set as the camera state.
        """
        if not image_url:
            raise RuntimeError("Image URL cannot be empty")

        logger.info(f"Publishing image URL {image_url} to {self._entity.url_topic}")
        self._state_helper(image_url, self._entity.url_topic)
